package.json

```diff
@@ -13,6 +13,7 @@
     "plugin"
   ],
   "scripts": {
+    "generate-types": "tsc",
```

Comment:
🟩 types check at root
🟩 type generation at package level 
🟥 type generation at root

Reply:
What is bad about generating types from the root?

Reply:
the output is on root that mean it's will not be used by the package

Reply:
No, it generates them in-place:

<img width="450" alt="image" src="https://github.com/user-attachments/assets/19aa3a65-bbc6-4cef-930f-250dd4de331b" />

```console
$ npm pack --dry-run

npm notice
npm notice 📦  @nodejs-loaders/tsx@1.0.0
npm notice Tarball Contents
npm notice 2.2kB README.md
npm notice 156B find-esbuild-config.d.mts
npm notice 175B find-esbuild-config.d.mts.map
npm notice 953B find-esbuild-config.mjs
npm notice 633B package.json
npm notice 803B tsx.d.mts
npm notice 143B tsx.d.mts.map
npm notice 1.9kB tsx.mjs
npm notice Tarball Details
npm notice name: @nodejs-loaders/tsx
npm notice version: 1.0.0
npm notice filename: nodejs-loaders-tsx-1.0.0.tgz
npm notice package size: 3.1 kB
npm notice unpacked size: 7.0 kB
```

Reply:
Hummm Okay in the past I doesn't do that.

But okay It's may help VS code. But IDK what is the interest of exporting type Nobody will want to import and do something of your function

Reply:
Ahh, but that's how I discovered VS Code wasn't picking them up:

```js
import * as aliasLoader from '@nodejs-loader/alias';

module.registerHooks(aliasLoader); // ← Angry TypeScript
```

Reply:
wait we can do that 🧐 ? _i always use the specifier_

Reply:
I believe you're thinking of [`module.register`](https://nodejs.org/api/module.html#moduleregisterspecifier-parenturl-options), which is different. This is [`module.registerHooks`](https://nodejs.org/api/module.html#moduleregisterhooksoptions)

I'm trying to switch to that in my `correct-ts-specifiers` codemod because of https://github.com/evanw/esbuild/issues/4016

Reply:
OOOOHHH okay let's do both (type gen + type check)